### PR TITLE
Clean makefile

### PR DESCRIPTION
**Pull request by SamLR**
Closes: #1477 

**What changed?**

* Added documentation for the tools used in the makefile
* Fixed bug due to missing plugins for `buf`
* General clean up

**Why?**

This should improve the developer experience and make understanding various workflows easier.

**How did you test it?**

As much as possible changes have been tested (albeit only on a mac)

### CODE REVIEW

**Review by ozamosi — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -75,6 +75,33 @@ To set up a development environment for the CLI
 6. Start the in-cluster API replacement job (powered by [http://tilt.dev](tilt.dev)) with `make cluster-dev`
 7. make or make unit-tests to ensure everything built correctly.
 
+### Requirements/tools
+
+This is a list of the tools you may need to install:
+
+* [go](go.dev) -- Primary compiler for the CLI.
```
Comment (dhwthompson) [inline README.md:82]: GitHub gets confused when URLs look like relative URLs.

```suggestion
* [go](https://go.dev) -- Primary compiler for the CLI.
```

File: README.md [hunk 2]
```diff
@@ -75,6 +75,33 @@ To set up a development environment for the CLI
...
+* [golangci-lint](https://github.com/golangci/golangci-lint/) -- A go linter.
+* [buf](https://buf.build/) -- To generate the protobufs used by the API.
+* [reflex](https://github.com/cespare/reflex) -- A file watcher.
+* [kind](https://kind.sigs.k8s.io/) -- Run kubernetes clusters in docker for testing.
+* [lcov](https://github.com/linux-test-project/lcov) -- Used for code coverage.
+
+Some other tools are installed automatically by the makefile for you:
+
+* [go-acc](https://github.com/ory/go-acc) -- Calculates code coverage for go.
+* [gcov2lcov](github.com/jandelgado/gcov2lcov) -- Converts output from go-acc to a format lcov understands.
```
Comment (dhwthompson) [inline README.md:95]:
```suggestion
* [gcov2lcov](https://github.com/jandelgado/gcov2lcov) -- Converts output from go-acc to a format lcov understands.
```

File: README.md [hunk 3]
```diff
@@ -75,6 +75,33 @@ To set up a development environment for the CLI
...
+* [controller-gen](sigs.k8s.io/controller-tools/cmd/controller-gen) -- Helps generate kubernetes controller code.
```
Comment (dhwthompson) [inline README.md:96]:
```suggestion
* [controller-gen](https://sigs.k8s.io/controller-tools/cmd/controller-gen) -- Helps generate kubernetes controller code.
```

File: README.md [hunk 4]
```diff
@@ -75,6 +75,33 @@ To set up a development environment for the CLI
 6. Start the in-cluster API replacement job (powered by [http://tilt.dev](tilt.dev)) with `make cluster-dev`
 7. make or make unit-tests to ensure everything built correctly.
 
+### Requirements/tools
```
Comment (dhwthompson) [inline README.md:78]: 🎉👍🧁 for extra documentation.

**Review by dhwthompson — COMMENTED** [review]
I don’t think I’m well-placed to review any of the front-end changes – that world is all still a bit of a mystery to me – but I can happily review the documentation side of things.

File: README.md (hunk 1 above)
Comment (SamLR) [reply README.md]: fixed in 7857b906

File: README.md (hunk 2 above)
Comment (SamLR) [reply README.md]: fixed in 7857b906

File: README.md (hunk 3 above)
Comment (SamLR) [reply README.md]: fixed in 7857b906

File: README.md (hunk 4 above)
Comment (SamLR) [reply README.md:78]: Danke

**Pull request merged**